src/utils/string.js

```diff
@@ -0,0 +1,13 @@
+/* @flow */
+
+export const replaceAtIndex = (str: string, index: number, char: string): string =>
+  `${str.substring(0, index)}${char}${str.substring(index + 1)}`;
+
+export const transformToEncodedURI = (string: string): string => {
+  const pattern = /\.\d/g;
+  let match;
+  while ((match = pattern.exec(string))) {
```

Comment:
Have a look at:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace

The `replace` function can receive a regexp, which will make the code way simpler.

Reply:
Done 👍